jabgui/src/main/java/org/jabref/gui/documentviewer/DocumentViewerViewModel.java

```diff
@@ -76,10 +80,25 @@ public StringProperty highlightTextProperty() {
     private void setCurrentEntries(List<BibEntry> entries) {
         if (entries.isEmpty()) {
             files.clear();
+            currentDocument.set(null); // Clear current document when no entries
         } else {
-            Set<LinkedFile> linkedFiles = entries.stream().map(BibEntry::getFiles).flatMap(List::stream).collect(Collectors.toSet());
-            // We don't need to switch to the first file, this is done automatically in the UI part
-            files.setValue(FXCollections.observableArrayList(linkedFiles));
+            Set<LinkedFile> pdfFiles = entries.stream()
+                                              .map(BibEntry::getFiles)
+                                              .flatMap(List::stream)
+                                              .filter(this::isPdfFile)  // 直接在这里过滤
+                                              .collect(Collectors.toSet());
+
+            if (pdfFiles.isEmpty()) {
+                // No PDF files found - clear the list and current document
+                files.clear();
+                currentDocument.set(null);
```

Comment:
I htink it would be good display a notification: dialogService.notify(Localization.lang("No pdf files  available"))